src/sdk/person/usePersonQuery.ts

```diff
@@ -0,0 +1,46 @@
+import { useEffect } from 'react'
+import { gql } from '@vtex/graphql-utils'
+import type {
+  PersonQueryQuery,
+  PersonQueryQueryVariables,
+} from '@generated/graphql'
+import { useSession } from '@faststore/sdk'
+
+import { useQuery } from '../graphql/useQuery'
+import type { QueryOptions } from '../graphql/useQuery'
+
+export const query = gql`
+  query PersonQuery {
+    person {
+      id
+      email
+      givenName
+      familyName
+      isAuthenticated
+    }
+  }
+`
+
+const usePersonQuery = (options?: QueryOptions) => {
+  const { data } = useQuery<PersonQueryQuery, PersonQueryQueryVariables>(
+    query,
+    {},
+    { ...options, fetchOptions: { ...options?.fetchOptions, method: 'POST' } }
```

Comment:
Does this request need to be a POST? I'm asking because the SWR (stale-while-revalidate) has no effect with POST.

Reply:
This has to be a POST just because it's sending the request to a GraphQL endpoint, isn't it?

Reply:
Actually, it has to be a POST to avoid caching

Reply:
Ok, great